Snowflake.Data.Tests/IntegrationTests/SFBindTestIT.cs

```diff
@@ -769,5 +774,290 @@ public void testPutArrayBind1()
+                        ++row;
+                        string faultMessage = $"Mismatch for row: {row}, {testCase}";
+                        Assert.AreEqual(row, reader.GetInt32(0));
+                        expected.IsEqual(reader.GetValue(1), comparisonFormat, faultMessage);
+                    }
+                }
+                Assert.AreEqual(1+smallBatchRowCount+bigBatchRowCount, row);
+            }
+            
+            void InsertSingleRecord(IDbConnection conn, string sqlInsert, DbType binding, int identifier, ExpectedTimestampValueProvider ts)
+            {
+                using (var insert = conn.CreateCommand(sqlInsert))
+                {
+                    // Arrange
+                    insert.Add("1", DbType.Int32, identifier);
+                    insert.Add("2", binding, ts);
+                    
+                    // Act
+                    s_logger.Info(sqlInsert);
+                    var rowsAffected = insert.ExecuteNonQuery();
+                    
+                    // Assert
+                    Assert.AreEqual(1, rowsAffected);
+                    Assert.IsNull(((SnowflakeDbCommand)insert).GetBindStage());
+                }
+            }
+
+            void InsertMultipleRecords(IDbConnection conn, string sqlInsert, DbType binding, int initialIdentifier, ExpectedTimestampValueProvider ts, int rowsCount, bool shouldUseBinding)
+            {
+                using (var insert = conn.CreateCommand(sqlInsert))
+                {
+                    // Arrange
+                    insert.Add("1", DbType.Int32, Enumerable.Range(initialIdentifier, rowsCount).ToArray());
+                    insert.Add("2", binding, ts, rowsCount);
+                    
+                    // Act
+                    s_logger.Debug(sqlInsert);
+                    var rowsAffected = insert.ExecuteNonQuery();
+                    
+                    // Assert
+                    Assert.AreEqual(rowsCount, rowsAffected);
+                    if (shouldUseBinding)
+                        Assert.IsNotEmpty(((SnowflakeDbCommand)insert).GetBindStage());
+                    else
+                        Assert.IsNull(((SnowflakeDbCommand)insert).GetBindStage());
+                }
+            }
+        }
+
+        private static string ColumnTypeWithPrecision(SFDataType columnType, Int32? columnPrecision) 
+            => columnPrecision != null ? $"{columnType}({columnPrecision})" : $"{columnType}";
+    }
+    
+    class ExpectedTimestampValueProvider
+    {
+        private readonly SFDataType _columnType;
+        private readonly DateTime? _expectedDateTime;
+        private readonly DateTimeOffset? _expectedDateTimeOffset;
+
+        internal static ExpectedTimestampValueProvider From(string timestampWithTimeZone, SFDataType columnType)
+        {
+            if (IsOffsetType(columnType))
+                return new ExpectedTimestampValueProvider(DateTimeOffset.ParseExact(timestampWithTimeZone, 
```

Comment:
personally I wouldn't inline the ParseExact method - splitting it into 2 steps would be more readable I think

Reply:
Fixed